### PR TITLE
Submit to nature address reviewers

### PR DESCRIPTION
**Pull request by Nikoleta-v3**
This `pr` aims to address the comments by the reviewers of Scientific Reports.

It includes a response letter.

Before re submitting:

- [x] @drvinceknight to approve the changes
- [ ] move to `Template for submissions to Scientific Reports`

### CODE REVIEW

File: main.tex [hunk 1]
```diff
@@ -342,84 +321,133 @@ \subsection{Best response memory-one strategies}\label{section:best_response_mem
...
+distribution of SSE is skewed to the left, indicating that the best response
+does exhibit ZDs behaviour and so could be extortionate, however, the best
+response is not uniformly a ZDs. A positive measure of skewness and kurtosis,
+and a mean of 0.34 indicate a heavy tail to the right. Therefore, in several
+cases the strategy is not trying to extort its opponents. More specifically,
+the best response memory-one with a theory of mind is being more adaptable than a ZDs.
+In~\cite{Hilbe2018} a similar behaviour is refereed to as the \textit{partners
+strategy}. Partners
+strategy aims to share the payoff for mutual cooperation, but it is ready
+to fight back when being exploited. In~\cite{Hilbe2018} they showed that partner strategies are
```
Comment (drvinceknight) [inline main.tex]:
```suggestion
to fight back when being exploited. In~\cite{Hilbe2018} it is shown that partner strategies are
```

File: main.tex [hunk 2]
```diff
@@ -342,84 +321,133 @@ \subsection{Best response memory-one strategies}\label{section:best_response_mem
...
+selected in evolutionary settings when populations are large and relationships
+stable and they allow for the evolution of cooperation. A best response memory-one strategy
```
Comment (drvinceknight) [inline main.tex]: Let's have a paragraph break here.

File: main.tex [hunk 3]
```diff
@@ -342,84 +321,133 @@ \subsection{Best response memory-one strategies}\label{section:best_response_mem
...
+in evolutionary settings has not been incorporated in the study of this manuscript, however,
+the best response memory-one strategy with self interactions can used to approximate
```
Comment (drvinceknight) [inline main.tex]:
```suggestion
the best response memory-one strategy with self interactions can used to approximate % TODO Vince to work on this a bit
```

Essentially we're saying:

- Best response mem1: adapt they are sometimes ZD and sometimes not;
- There is a paper by Hilbe that described a strategy designed to do the same thing and was evolutionary stable.
- That's cool.

File: main.tex [hunk 4]
```diff
@@ -342,84 +321,133 @@ \subsection{Best response memory-one strategies}\label{section:best_response_mem
...
     \label{fig:behaviour_violin_plots}
 \end{figure}
 
-\subsection{Longer memory best responses}
-
-This section focuses on the memory size of strategies. The effectiveness of
-memory in the IPD has been previously explored in the literature, however, no one
-has compared the performance of longer-memory
-strategies to memory-one best responses.
+The second limitation this paper argues that memory-one extortionate strategies
```
Comment (drvinceknight) [inline main.tex]: This sentence needs fixing.

File: main.tex [hunk 5]
```diff
@@ -38,8 +38,8 @@
 \newtheorem{lemma}[theorem]{Lemma}
 \def\arraystretch{1.5}
 
-\title{Stability of defection, optimisation of strategies and the limits of
-       memory in the Prisoner's Dilemma.}
+\title{Best response memory-one strategies with a theory of mind: the limitations
+of extortion and restricted memory.}
```
Comment (drvinceknight) [inline main.tex]:
```suggestion
\title{A theory of mind: Best responses to memory-one strategies. The limitations
of extortion and restricted memory.}
```

File: main.tex [hunk 6]
```diff
@@ -477,14 +504,13 @@ \subsection{Longer memory best responses}
 
 \section{Discussion}
```
Comment (drvinceknight) [inline main.tex]: Let's include the stability of defection more prominently here.

```
One of the main results of this paper is a closed form for the utility of a strategy (blablabla) this not only allows for the analysis we've discussed but can also be used to prove other theoretic results such as

THEORM

PLOT
```

Comment (drvinceknight) [reply main.tex]: Put this where you are currently discussing it.

File: response/main.tex [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+\author{Nikoleta E. Glynatsi, Vincent A. Knight}
+ 
+\begin{document}
+
+\maketitle
+
+We would like to open this response by thanking the reviewers for their
+thoughtful comments and suggestions. We have fully taken their comments on board
+and made the necessary changes.
```
Comment (drvinceknight) [inline response/main.tex]:
```suggestion
and made significant modifications to both the mathematical arguments and the narrative. We feel this has greatly improved the work.
```

File: response/main.tex [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+The main modifications include the improvement of the theorems' proofs and the
```
Comment (drvinceknight) [inline response/main.tex]:
```
Both reviewers commented on different aspects of the paper and so our modifications naturally fit in to two categories (which we will discuss in detail comment by comment):

- The first is an improvement of the theoretic results both in terms of their proofs but also the result
 themselves. For example, we omitted to include a detail about the derivative of quadratic form in one 
 of our proofs and also had only given that quadratic form for the specific case of (R, P, T, S)=(). 
 The reviewer was very right to point out issues like this and we have addressed them all.
- The second reviewer questioned the general narrative of the paper as well as the evolutionary work. 
 This was very helpful and we took time to reflect on the narrative of the paper. As a result of this we 
 have changed the title to TITLE and reorganised some of the content. We have also clarified the 
 nature of the evolutionary process we're discussing.

We will now take each comment of the reviewers in turn and highlight our efforts made to improve the work:
```

File: response/main.tex [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+\section{Response}
+
+\textbf{Regarding the comments of Reviewer 1:}
+
+\begin{verbatim}
+     ``Theorem 2: First, it is not clear how equation 17 is obtained.''
+\end{verbatim}
+
+We have added further details on how to obtain the equation in the proof of Theorem 2
```
Comment (drvinceknight) [inline response/main.tex]: Can we for each of these, add a bit more detail.

So for example, here we'd say:

```
This comment point out that we had skipped the step of multiplying the steady state vector by the utilities \(v\dot (RPTS)\). This has been added in the proof of what used to be T 2 but is now T1.
```

File: response/main.tex [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+\begin{verbatim}
+    ``Theorem 2: Moreover, even if that's correct, it is only for a
+    specific PD payoff matrix. One cannot state the general results in (4) from
+    just a specific case. To obtain (4), authors need to establish it for the
+    general PD payoff matrix.''
+\end{verbatim}
+
+We agree with the comment of the reviewer. The utility was
+specific only for a given payoff matrix. We have generalised our result by
+considering the generalised payoffs \((R, S, T, P)\).
```
Comment (drvinceknight) [inline response/main.tex]:
```suggestion
considering the generalised payoffs \((R, S, T, P)\) and feel that not only is this a more useful 
result and whilst we were worried that this closed form could be very messy it is in fact relatively elegant.
```

File: response/main.tex [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+
+\begin{verbatim}
+     ``Theorem 3: The proof in the Appendix did not show how to obtain equations
+    (22) - (24), the key results of the Theorem''
+\end{verbatim}
+
+We have addressed this comment by making additions to the proof.
```
Comment (drvinceknight) [inline response/main.tex]: Add a couple of details.

File: response/main.tex [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+
+\begin{verbatim}
+     ``Theorem 3: and what was stated in (5)''
+\end{verbatim}
+
+We have addressed this comment by specifying that it is an algebraic manipulation
```
Comment (drvinceknight) [inline response/main.tex]:
```
We have added a sentence clarifying that this is obtained by taking the average over the opponents of the previously obtained expression for the utility.
```

File: response/main.tex [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+    hard to evaluate the impact of the findings, or give any hope for reproducing
+    the results. I realize that there is a repository containing raw data, but
+    the lack of explanation of the setup really harms any expectation of the
+    paper being self-contained. The issue of the evolutionary process not being
+    explained also prevented me from fully following the argument the authors
+    make for why best responses in the evolutionary process would be less
+    extortionate. ''
+\end{verbatim}
+
+We agree with the comment of the reviewer. We are not running a evolutionary
```
Comment (drvinceknight) [inline response/main.tex]:
```suggestion
We agree with the comment of the reviewer. We are not running a evolutionary % TODO Vince to modify
```

File: response/main.tex [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+and it's results.
+
+\begin{verbatim}
+    ``Another related problem is that the authors don't go into more detail about
+    their SSE method. A more expansive explanation (also of Table 1) would be of
+    critical importance, seeing that this is one of the central points of the
+    results section.''
+\end{verbatim}
+
+We have address this comment by including details of the SSE method and a better
```
Comment (drvinceknight) [inline response/main.tex]:
```suggestion
We have included more details of the SSE method and a better
```

File: response/main.tex [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+    defection) or "rival" (defecting/extortionate) strategies evolve in the IPD. It
+    seems that the authors of the current work investigate a similar distinction
+    (however, they should explain what exactly they mean with "adaptable", as it
+    features as a contrast to "extortionate"), and thus would profit from discussing
+    the Hilbe et al. article as well.
+    ''
+\end{verbatim}
+
+We thank the reviewer for their article recommendation. The findings of Hilbe et
+ al and how our results relate to them, are discussed in the ``Results'' section.
```
Comment (drvinceknight) [inline response/main.tex]:
```suggestion
 al and how our results relate to them, are discussed in the ``Results'' section. % TODO Vince to add details
```

File: response/main.tex [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+    now, the paper reads a bit like a list of results that are more or less loosely
+    tied together (especially Section 1.2 suffers from this problem in comparison to
+    the others). It would be good to motivate single sections in the broader context
+    of the message the paper wants to tell. ''
+\end{verbatim}
+
+Thank you for this comment which encouraged us to reflect on the narrative of the
+paper, as a result we have modified the title of the manuscript and the titles
+of the sections. Moreover, the results of Section 1.2 have been to the Appendix,
+and the Theorem 1 (old version Theorem 2) has been moved in the main body.
```
Comment (drvinceknight) [inline response/main.tex]:
```
We feel that this...
```

File: main.tex [hunk 17]
```diff
@@ -342,84 +321,133 @@ \subsection{Best response memory-one strategies}\label{section:best_response_mem
...
+\begin{figure}[!htbp]
+    \begin{subfigure}{0.47\textwidth}
+        \begin{center}
+            \includegraphics[width=\linewidth]{img/best_respones_sserror.pdf}
+        \end{center}
+        \caption{SEE distribution for best response in tournaments without self interactions.}
+    \end{subfigure}\hfill
+    \begin{subfigure}{0.47\textwidth}
+        \begin{center}
+            \includegraphics[width=\linewidth]{img/evo_sserror.pdf}
```
[the remark this answers is not in the document]
Comment (Nikoleta-v3) [reply main.tex]: https://github.com/Nikoleta-v3/Memory-size-in-the-prisoners-dilemma/issues/57

**Comment by Nikoleta-v3** [comment]
@drvinceknight I have now addressed all the comments.

Note that:
-  b349ef4 

and I have opened the following issues:

- https://github.com/Nikoleta-v3/Memory-size-in-the-prisoners-dilemma/issues/57
- https://github.com/Nikoleta-v3/Memory-size-in-the-prisoners-dilemma/issues/58

**Pull request merged**